### ISSUE TITLE
Training won't start

### ISSUE THREAD

**Issue by akas0018** [TristaZeng/ZS-DeconvNet#4]
Hi There, I find this tool really useful for my research. I am having issues starting the training on the FIJI plugin though. I get the following error, for both 2D and 3D trainings. I would be really grateful for any help. 
Thanks
![tempImagews6d97](https://github.com/TristaZeng/ZS-DeconvNet/assets/113228877/ff281195-72dc-452e-a107-228b7b32d629)

**Comment by TristaZeng** [comment]
Hi, please tell us what you did before the error in detail so that we can repeat your doing to locate the problem.

**Comment by akas0018** [comment]
I just tried to train a model on the opened image and specified the path to PSF and clicked on train and then this. I am now trying to get the code working through terminal but ImageJ plugin is a lot more convenient.

**Comment by TristaZeng** [comment]
May be due to some problems with Fiji version, CUDA version or something alike. Since we cannot repeat your error, you can try using this Fiji app with ZS plugin to replace your Fiji: https://drive.google.com/file/d/1IoAVnrMHXtuXzXbFXvChFZFk_YCDmbCt/view?usp=sharing

**Comment by akas0018** [comment]
Thanks a lot. I am now getting a slightly different error with the fiji installation you shared
![Uploading tempImagezfYPCY.gif…]()

**Comment by akas0018** [comment]
The problem seems to be the cuda version mismatch. Fiji will only let me install upto cuda10 :(
<img width="1028" alt="Screenshot 2023-11-19 at 5 17 36 pm" src="https://github.com/TristaZeng/ZS-DeconvNet/assets/113228877/f1219093-5255-41ca-baaf-b4961ad94e8d">

**Comment by TristaZeng** [comment]
Yes, that is a problem...The package we built our plugin on can only support CUDA<=10.0. You can only try with graphic cards such as RTX 2080 or python.

**Comment by akas0018** [comment]
So I downloaded the [3D time-lapsing data_LLSM_Mitosis_Mito.zip] that has the PSF and 1000 images in the training dataset. I created a separate directory for test dataset using 50 of the training dataset and tried to start training and got the following error. Could you please have a look.  Thanks a lot for all your help
![Screenshot from 2023-11-20 21-03-46](https://github.com/TristaZeng/ZS-DeconvNet/assets/113228877/03517786-5568-4a76-ade4-5527c84796d8)

![Screenshot from 2023-11-20 21-05-52](https://github.com/TristaZeng/ZS-DeconvNet/assets/113228877/05ff87f4-2c09-4af8-b901-d7feabac5e2c)

**Comment by TristaZeng** [comment]
This error occurs due to the specified training data folder `/home/superviser/Aseem/analysis/zs_deconv/ZS-DeconvNet/trail/4Ddata_LLSM_Mitosis_Mito/train_data/input` being empty. I guess you placed 950 stacks inside the folder `/home/superviser/Aseem/analysis/zs_deconv/ZS-DeconvNet/trail/4Ddata_LLSM_Mitosis_Mito/train_data.` You should go one step further and use the MATLAB codes to generate `/home/superviser/Aseem/analysis/zs_deconv/ZS-DeconvNet/trail/4Ddata_LLSM_Mitosis_Mito/train_data/input` and `/home/superviser/Aseem/analysis/zs_deconv/ZS-DeconvNet/trail/4Ddata_LLSM_Mitosis_Mito/train_data/gt` from the 950 raw data (see **3. Training dataset generation** in the readme file). This data generation step is included in the Fiji plugin but should be done in advance in MATLAB if using Python. Remember to change the default output path of MATLAB codes, and check the `dx,` `dz,` `dxpsf,` `dzpsf` parameters in Python (dx, dz properties are embedded in the data TIFF file).

**Comment by akas0018** [comment]
Thanks a lot. I was able to get the training done for complete 10000 epochs.  I seem to be getting an error when running inference. I tried running it both on the matlab generated ground truths as well as on raw data
![inference](https://github.com/TristaZeng/ZS-DeconvNet/assets/113228877/0ebf031d-90a0-482f-970e-f64344c62157)

**Comment by TristaZeng** [comment]
That is very likely due to the number of stacks in your test data path being larger than the length of parameters `num_seg_window_z`, `overlap_z`, `num_seg_window_x`, `overlap_x`, `num_seg_window_y` or `overlap_y`. For example, if you have 2 stacks in `input_dir`, and you want to segment the first stack into 2 parts, the second stack into 3 parts axially during inference, with both stacks having overlap of 4 pixels along the z axis, you should set `num_seg_window_z=[2,3], overlap_z=[4,4]`. The same goes for other parameters. By the way, inference should be done on raw data.

**Comment by akas0018** [comment]
Hi, Thanks a lot for all your help. I have been able to do training and inference on LLSM data released with your paper. The results are as stunning as claimed in your manuscript. I am however only doing the inference on a single slice at a time. Can you tell me how I should modify the inference code below to be able to run inference on a few hundred slices at a time?
![Screenshot from 2023-11-24 16-16-02](https://github.com/TristaZeng/ZS-DeconvNet/assets/113228877/565db396-edb5-484d-93d2-4866196709d9)


I am trying to train this model on my own data as well (3D SIM). I am however unable to generate the training patches for it using the matlab script. what should be the input for the Create_corrupt_img_3D.m script. When I provide the 3D .mrc files, it generates three different folders Snr 01/02/03 but they are all empty. I only have a single channel in my SIM dataset, Does the script require multiple channels? What parameters should I change in the .m file to make it work? I think the issue here will be the code not being able to read the mrc file properly / meta data being in a different format. Could you tell me what paramters should I directly input in the .m script to prevent it from trying to read the metadata (eg Nx, Ny), Not sure what header(46) is.
<img width="1498" alt="Screenshot 2023-11-28 at 3 05 30 pm" src="https://github.com/TristaZeng/ZS-DeconvNet/assets/113228877/c59b1d9a-5b15-4420-88ac-49c150924b5d">

I am attaching a sample file at this link : https://drive.google.com/file/d/1tO_4e8xIapP284h1tX_UbENRCEtNfg9U/view?usp=share_link

**Comment by TristaZeng** [comment]
On a second thought, the program did not even throw a warning which means most likely `file_count=0`. Check your `root_dir` and other settings. 
And we seem to have 3 channels because we have 3 SNR levels, since you have only one you can delete the outer loop or set `id_max=1`.
